### PR TITLE
Replace non-finite values and null-fill across timerange

### LINKED ISSUE #68702: SparklineCell: freezes browser if there's only one value in data frame
If sparkline data frame contains a single value, it can freeze browser because uPlot gets into an infinite loop here: https://github.com/leeoniya/uPlot/blob/91b459fbcb86d39d1e7dfafd16ee5db9872987f7/dist/uPlot.esm.js#L1328

Setting `sparkline.timeRange` would fix it,  but time range is currently not available for table panel cells.

Also: should probably null-fill sparkline data. Currently if there's data not for the entire time range, sparkline will misleadinly render across entire X axis, as illustrated below:

image

### PR DESCRIPTION
**Pull request by mdvictor**
Replaces non-finite values (e.g: NaN, +/- Infinity) and null fills the sparkline data over the entire time-range. Not sure if this is the best approach, feedback appreciated :D

Fixes #68702 (the part regarding null fill)

- [x] Fix code so it doesn't break tests

### CODE REVIEW

File: packages/grafana-ui/src/components/Table/SparklineCell.tsx [hunk 1]
```diff
@@ -40,6 +40,20 @@ export const SparklineCell = (props: TableCellProps) => {
...
   }
 
+  // Get the step from the first two values to null-fill the x-axis based on timerange
+  if (sparkline.x && sparkline.x.values.length > 1) {
+    sparkline.x.config.interval = sparkline.x.values[1] - sparkline.x.values[0];
+  }
+
+  // Remove non-finite values, e.g: NaN, +/-Infinity
+  sparkline.y.values = sparkline.y.values.map((v, i) => {
+    if (!isFinite(v)) {
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Table/SparklineCell.tsx:50]: Yo! I didn't even know about `isFinite()`; nice!

Comment (leeoniya) [reply packages/grafana-ui/src/components/Table/SparklineCell.tsx:50]: you should use `Number.isFinite()` tho ;)

Comment (baldm0mma) [reply packages/grafana-ui/src/components/Table/SparklineCell.tsx:50]: Ah, due to the implicit type conversion of non-Numbers? Nice catch. This guy JavaScripts.

**Comment by leeoniya** [comment]
would like to test this locally, but not sure how to make a `SparklineCell` dataframe with `gdev-testdata`.

can you plz attach debug dashboard or (better) include a gdev dashboard that contains nulls and NaNs, and less data than the time range?

**Review by leeoniya — APPROVED** [review]

File: packages/grafana-ui/src/components/Table/SparklineCell.tsx [hunk 2]
```diff
@@ -40,6 +40,20 @@ export const SparklineCell = (props: TableCellProps) => {
     );
   }
 
+  // Get the step from the first two values to null-fill the x-axis based on timerange
+  if (sparkline.x && !sparkline.x.config.interval && sparkline.x.values.length > 1) {
+    sparkline.x.config.interval = sparkline.x.values[1] - sparkline.x.values[0];
+  }
+
+  // Remove non-finite values, e.g: NaN, +/-Infinity
+  sparkline.y.values = sparkline.y.values.map((v, i) => {
```
Comment (baldm0mma) [inline packages/grafana-ui/src/components/Table/SparklineCell.tsx:49]: Is the index arg here necessary? Maybe either remove it or underscore it?

**Review by baldm0mma — APPROVED** [review]
One small nit, but LGTM!

**Pull request merged**